### PR TITLE
Add Cucumber tests

### PR DESCRIPTION
**Pull request by SebastienTainon**
- Add Cucumber.js BDD test framework
- Add hooks for tests (db setup, server setup)
- Add tests for endpoint /tasks/{taskId}
- Add CircleCI config

### CODE REVIEW

File: features/steps/task_steps.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import {testServer} from '../support/hooks';
+
+interface TaskStepsContext {
+    defaultTaskId: string,
+    response: ServerInjectResponse,
+}
+
+
+Given(/^there is a default task in the database$/, async function (this: TaskStepsContext) {
+    const taskId = getRandomId();
```
Comment (smadbe) [inline features/steps/task_steps.ts:17]: It is usually better not to rely on any randomness in tests (or a least to mock the random function to have a fixed value)

**Review by smadbe — CHANGES_REQUESTED** [review]

File: features/get_task.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Feature: Get task
+
+  Scenario: Get task by id
+    Given there is a default task in the database
+    When the server receives a GET request to "/tasks/{defaultTaskId}"
+    Then the server should return the status code 200
+    Then the server response should include the following object: "get_task_response.json"
```
[the remark this answers is not in the document]
Comment (SebastienTainon) [reply features/get_task.feature]: I've changed to a static Gherkin datatable, it's indeed more readable, thanks! :+1:

File: features/steps/task_steps.ts [hunk 3]
```diff
@@ -1,95 +1,28 @@
...
-Then(/^the server response should include the following object: "([^"]*)"$/, async function (this: TaskStepsContext, fileName: string) {
-    expect(this.response.statusCode).to.equal(200);
-
+Then(/^the response body should be: "([^"]*)"$/, async function (this: TaskStepsContext, fileName: string) {
     const expectedResponse = JSON.parse(await readFile(path.join(__dirname, '..', fileName), "utf8"));
     const payload = JSON.parse(this.response.payload);
-
-    expect(payload.id).to.equal(this.defaultTaskId);
-    expect(payload).containSubset(expectedResponse);
+    console.log(payload);
```
Comment (smadbe) [inline features/steps/task_steps.ts:22]: debugging leftover?

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**